### ISSUE TITLE
iOS - Master Detail Page always displays

### ISSUE THREAD

**Issue by vsfeedback** [xamarin/Xamarin.Forms#8005]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/763810/ios-master-detail-page-always-displays.html)._

---
<p>We are using a MasterDetailPage for a hamburger menu on an iPad app with a binding of the IsPresented property to a variable of the associated class. When running the app on a simulator on iOS 13, the menu is always visible and cannot be hidden even when hardcoding IsPresented to false. If I run the code on an iPad on iOS 12 the menu works as expected and is not presented by default. Are there any code changes needed when using the MasterDetailPage component as a result of iOS 13?</p>

---
#### Visual Studio Feedback System on 10/8/2019, 00:24 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### todd.mcmullen on 10/8/2019, 05:25 AM: 

<p>Same issue here. iOS 13.1.2 (and 13.2 beta) with XCode 11 (not 11.1 - app store keeps erroring out).</p><p>Debugging shows that IsPresented = true, setting it in code or through the immediate window does absolutely nothing. It's in PopOver mode if that matters. </p><p>Nothing I do seems to affect the presentation and the gesture to close it doesn't work either. </p>

#### todd.mcmullen on 10/8/2019, 05:29 AM: 

<p>Also, it's like popover mode doesn't work. My details page isn't 'under' the master/menu page. They're on the same layer. Detail view is 'squished' to the right. (super technical)</p>

#### quyen.castellanos on 10/8/2019, 00:38 PM: 

<p>Update: This issue is also happening on an iPad on iPadOS 13.1.2, when deploying from the latest version of VS for Mac. Launching a released version of the the application from TestFlight on the same iPad (13.1.2), the issue does not exist. I believe because the build was generated on an earlier version of Xamarin.</p>

---
### Original Solutions
(no solutions)

**Comment by chrisweihu** [comment]
HI Xamarin,
Any update on this bug? I also ran into this bug as well. The master side menu doesn't hide when in Popover mode. This bug is reproducible creating a new Xamarin.Form project using the MasterDetailPage Project template from VisualStudio.

**Comment by samhouts** [comment]
@vsfeedback  Can you please attach a small project that demonstrates this issue? Thanks!

**Comment by chrisweihu** [comment]
[TestMasterDetail.zip](https://github.com/xamarin/Xamarin.Forms/files/3741499/TestMasterDetail.zip)

Here is the Xamarin.Form project using the MasterDetailPage Project template from VisualStudio.

Note that: MasterBehavior = MasterBehavior.Popover;  but the Master Sidebar menu always show, can't be hidden.

The only workaround is to overwrite the MasterDetailPage  renderer with **"this.PreferredDisplayMode = UISplitViewControllerDisplayMode.PrimaryOverlay;"**

```
using UIKit;
using Xamarin.Forms;
using Xamarin.Forms.Platform.iOS;
[assembly: ExportRenderer(typeof(MasterDetailPage), typeof(xxx.Mobile.Renderers.IPadMasterDetailRenderer), UIUserInterfaceIdiom.Pad)]
namespace xxx.Mobile.Renderers
{
    class IPadMasterDetailRenderer : TabletMasterDetailRenderer
    {
        public override void ViewDidLoad()
        {
            base.ViewDidLoad();

            this.PreferredDisplayMode = UISplitViewControllerDisplayMode.PrimaryOverlay;
        }
    }
}
```

**Comment by kingces95** [comment]
I'm able to expand and collapse the menu on master. Can you provide steps to go thru using the reproduction and the expected and actual behavior?

![188EF283-9B9B-4F90-8BA4-1A1D511B4514](https://user-images.githubusercontent.com/4120386/67133815-4c660700-f1aa-11e9-8c06-15f5c6afef7d.GIF)

**Comment by qucas** [comment]
@kingces95 @samhouts It looks like this issue still exists in the latest version. Where can I find an older version (pre-8.3) of VS for Mac? Any download link I find for earlier versions navigates me to the latest download page and I can't create builds of the app with this issue.

**Comment by samhouts** [comment]
@qucas I don't believe that older versions are available for download. Does the workaround in https://github.com/xamarin/Xamarin.Forms/issues/8005#issuecomment-543385160 work for you? Thanks!
